### PR TITLE
pkg-config: always pass --static in static builds

### PR DESCRIPTION
**Pull request by alyssais**
By default, pkg-config output will not include Requires.private dependencies (which specify dependencies that only apply when building statically).  It will only do this when passed --static.  In pkgsStatic, let's ensure pkg-config is always operating in static mode, because pkgsStatic will ensure that any libraries pkg-config might find will always be static, and so will always need their Requires.private dependencies.

This is very useful for Meson builds, because otherwise Meson will only pass --static to pkg-config if the dependency was explicitly `"static : true"`, which is not likely to be the case for most stuff we're building.

An alternative proposal was to patch Meson to` add a spe`cial environment variable to force dependencies to be `"static : true"`.  I feel that the approach I've taken here is less invasive.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/build-support/pkg-config-wrapper/pkg-config-wrapper.sh [hunk 1]
```diff
@@ -12,10 +12,12 @@ if [ -z "${NIX_PKG_CONFIG_WRAPPER_FLAGS_SET_@suffixSalt@:-}" ]; then
...
 fi
 
+set -- "$@" @addFlags@
+
 if (( ${#role_suffixes[@]} > 0 )); then
     # replace env var with nix-modified one
     PKG_CONFIG_PATH=$PKG_CONFIG_PATH_@suffixSalt@ exec @prog@ "$@"
 else
-    # pkg-config isn't a bonafied dependency so ignore setup hook entirely
+    # pkg-config isn't a bona fide dependency so ignore setup hook entirely
```
Comment (Ericson2314) [inline pkgs/build-support/pkg-config-wrapper/pkg-config-wrapper.sh:21]: :blush: thanks

**Review by Ericson2314 — APPROVED** [review]

File: pkgs/build-support/pkg-config-wrapper/pkg-config-wrapper.sh (hunk 1 above)
Comment (Mic92) [inline pkgs/build-support/pkg-config-wrapper/pkg-config-wrapper.sh:21]: easier to understand for non-natives without having to look in the dictionary:

```suggestion
    # pkg-config isn't a real dependency so ignore setup hook entirely
```

File: pkgs/build-support/pkg-config-wrapper/default.nix [hunk 2]
```diff
@@ -47,6 +47,8 @@ stdenv.mkDerivation {
   dontBuild = true;
   dontConfigure = true;
 
+  addFlags = lib.optional stdenv.targetPlatform.isStatic "--static";
```
Comment (Mic92) [inline pkgs/build-support/pkg-config-wrapper/default.nix:50]:
```suggestion
  # additional command line flags passed to patchelf
  addFlags = lib.optional stdenv.targetPlatform.isStatic "--static";
```

**Review by Mic92 — APPROVED** [review]
untested but makes sense.

**Review by jonringer — APPROVED** [review]
LGTM

we can stabilize in staging-next

**Pull request merged**